packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts

```diff
@@ -0,0 +1,63 @@
+import {HealthCheckInterface} from '../../types';
+import fs from 'fs';
+import {promisify} from 'util';
+import {findProjectRoot} from '@react-native-community/cli-tools';
+
+const applePlatformFolders = ['ios', 'macos'];
+const xcodeEnvFile = '.xcode.env';
+
+function allAppleProjectsHaveXcodeEnv(projectRoot: string): boolean {
+  return applePlatformFolders
+    .map((folder) => xcodeEnvFileExistsFor(folder, projectRoot) ?? true)
+    .reduce((previousValue, currentValue) => previousValue && currentValue);
+}
+
+/**
+ * Checks whether the basePath+platform folder exists. If that happens, it means
+ * that the project supports that platform.
+ * Returns whether the file exists in a supported platform. It returns `null` if
+ * the platform is not supported
+ * @param platform the platform to be checked
+ * @param basePath the path to the parent of the platform
+ * @returns If the platform is supported, `true` if there is the file, `false` otherwise. If the platform is not supported, `null`
+ */
+function xcodeEnvFileExistsFor(
+  platform: string,
+  basePath: string,
+): boolean | null {
+  const platformPath = basePath + platform;
+  if (fs.existsSync(platformPath)) {
+    return fs.existsSync(platformPath + xcodeEnvFile);
+  }
+  return null;
+}
+
+export default {
+  label: '.xcode.env',
+  description: 'File to customize Xcode environment',
+  getDiagnostics: async () => {
+    const projectRoot = findProjectRoot();
+    const allExists = allAppleProjectsHaveXcodeEnv(projectRoot);
+    return {
+      needsToBeFixed: !allExists,
+    };
+  },
+  runAutomaticFix: async () => {
+    const templateFilePath = '_xcode.env';
+    const projectRoot = findProjectRoot();
+
+    const src =
+      projectRoot +
+      '/node_modules/react-native/template/ios/' +
+      templateFilePath;
```

Comment:
This is assuming the installation path of `react-native`. We should probably use `require.resolve` instead?

Reply:
I hope I used it properly. Let me know if I have to update the call.